### PR TITLE
[Portfolio] Fix free balance parsing 

### PR DESCRIPTION
**Pull request by Herklos**
Depends on https://github.com/Drakkar-Software/OctoBot-Trading/pull/116

### CODE REVIEW

**Comment by Herklos** [comment]
WIP adding tests

**Comment by Herklos** [comment]
> WIP adding tests

Done

File: octobot_trading/data/order.py [hunk 1]
```diff
@@ -100,10 +100,6 @@ def update(self, symbol, order_id="", status=OrderStatus.OPEN,
             self.origin_price = price
             changed = True
 
-        # TODO
-        # if fee is not None and self.fee != fee and self.filled_quantity is not None:
-        #     self.fee = self.get_computed_fee()
-
```
Comment (GuillaumeDSM) [inline octobot_trading/data/order.py]: Will be updated in https://github.com/Drakkar-Software/OctoBot-Trading/pull/110/files
into: 
```python
        if fee is not None and self.fee != fee:
            self.fee = fee
```
(fee is used in parsed historical trades)

File: octobot_trading/data/portfolio.py [hunk 2]
```diff
@@ -56,11 +56,13 @@ def __init__(self, exchange_name, is_simulated=False):
...
-        else balance[currency][PORTFOLIO_AVAILABLE],
-                                     PORTFOLIO_TOTAL: balance[currency][CONFIG_PORTFOLIO_TOTAL]}
-                          for currency in balance}
+        self.portfolio = {
+            currency: {
+                PORTFOLIO_AVAILABLE: balance[currency][CONFIG_PORTFOLIO_FREE]
+                if CONFIG_PORTFOLIO_FREE in balance[currency] else balance[currency][PORTFOLIO_AVAILABLE],
+                PORTFOLIO_TOTAL: balance[currency][CONFIG_PORTFOLIO_TOTAL]
+                if CONFIG_PORTFOLIO_TOTAL in balance[currency] else balance[currency][PORTFOLIO_TOTAL]}
+            for currency in balance}
```
Comment (GuillaumeDSM) [inline octobot_trading/data/portfolio.py]: 👍

File: octobot_trading/exchanges/exchange_manager.py [hunk 3]
```diff
@@ -16,12 +16,12 @@
 import uuid
 
 from octobot_channels.util.channel_creator import create_all_subclasses_channel
-
 from octobot_commons.config_util import has_invalid_default_config_value
 from octobot_commons.constants import CONFIG_ENABLED_OPTION
 from octobot_commons.enums import PriceIndexes
 from octobot_commons.logging.logging_util import get_logger
 from octobot_commons.timestamp_util import is_valid_timestamp
+
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/exchange_manager.py]: Why this newline ?

File: octobot_trading/exchanges/rest_exchange.py [hunk 4]
```diff
@@ -13,17 +13,16 @@
...
 
 import ccxt.async_support as ccxt
 from ccxt.async_support import OrderNotFound, BaseError, InsufficientFunds
 from ccxt.base.errors import ExchangeNotAvailable, InvalidNonce, BadSymbol
-
 from octobot_commons.config_util import decrypt
 from octobot_commons.constants import MSECONDS_TO_MINUTE
 from octobot_commons.dict_util import get_value_or_default
 from octobot_commons.enums import TimeFramesMinutes
+
```
Comment (GuillaumeDSM) [inline octobot_trading/exchanges/rest_exchange.py]: Why this newline ?

File: tests/test_utils/order_util.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    await limit_or_stop_order.update_order_status(last_prices)
+
+
+async def fill_market_order(market_order, price):
+    last_prices = [{
+        "price": price
+    }]
+
+    await market_order.update_order_status(last_prices)
```
Comment (GuillaumeDSM) [inline tests/test_utils/order_util.py]: 👍

**Pull request merged**